gallery/tutorials/tutorials/micrograph_source.py

```diff
@@ -0,0 +1,220 @@
+    micrograph_size=500,
+    micrograph_count=4,
+    seed=1234,
+)
+
+# Plot the micrographs
+noisy_micrograph.micrographs[:].show()
+
+# %%
+# We can also plot the un-noisy micrographs using the ``clean_micrographs`` accessor
+noisy_micrograph.clean_micrographs[:].show()
+
+# %%
+# Interparticle Distance
+# ----------------------
+# By default, particle distance is set to avoid collisions.
+# We can use this argument to control the minimum distance between particle centers
+# However, setting this argument too large may generate insufficient centers.
+
+# Let's increase the number of particles to show overlap.
+# Create a new simulation to meet the minimum required amount of projections.
+n_particles_per_micrograph = 20
+
+colliding_sim = Simulation(
+    L=64,
+    n=n_particles_per_micrograph * n_micrographs,
+    C=1,
+    amplitudes=1,
+    offsets=0,
+    unique_filters=ctfs,
+)
+
+
+# Set the interparticle distance to 1, which adds at least one pixel of separation between centers.
+# This allows particles to collide but not completely overlap.
+colliding_micrograph = MicrographSource(
+    colliding_sim,
+    interparticle_distance=1,
+    noise_adder=noise,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+)
+
+# Plot the micrographs
+colliding_micrograph.micrographs[:].show()
+
+# %%
+# Boundary
+# --------
+# By default, the boundary is set to half of the particle width , which will completely contain every particle inside the micrograph.
+# Setting ``boundary=0`` will allow particle centers to generate along the edges.
+# Positive values (measured in pixels) move the boundaries inward, while negative values move the boundaries outward.
+
+# Create a micrograph with a negative boundary, allowing particles to generate outward.
+outbound_micrograph = MicrographSource(
+    colliding_sim,
+    boundary=-20,
+    interparticle_distance=1,
+    noise_adder=noise,
+    particles_per_micrograph=n_particles_per_micrograph,
+    micrograph_size=500,
+    micrograph_count=n_micrographs,
+)
+
+# Plot the micrographs
+outbound_micrograph.micrographs[:].show()
+
+# %%
+# Particle IDs
+# ------------
+# Each particle comes from the simulation, and has its own ID relative to its ``Simulation`` and micrograph.
+# Let's choose 4 random numbers, as our global (Simulation) particle IDs from ``test_micrograph=1``.
+test_micrograph = 1
+local_particle_ids = np.random.choice(n_particles_per_micrograph, 4)
+print(f"Local particle IDS: {local_particle_ids}")
+
+# %%
+# We can obtain the images from our MicrographSource by retrieving their centers and plotting the boundary boxes.
+centers = np.zeros((len(local_particle_ids), 2), dtype=int)
+for i in range(local_particle_ids.shape[0]):
+    centers[i] = colliding_micrograph.centers[test_micrograph][local_particle_ids[i]]
+
+# Let's use the particles' centers and sizes to perform "perfect particle picking" on this test micrograph.
```

Comment:
We should consider making this a function of the source, returning a `CoordinateSource` that represents a “perfect particle picking” algorithm as you describe it. Maybe make an issue?

Reply:
The only concern is when the particle goes outside the micrograph, but I can make it an issue right now :)